cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.linkedin.kafka.cruisecontrol.async.OperationFuture;
+import com.linkedin.kafka.cruisecontrol.common.KafkaCruiseControlThreadFactory;
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.apache.kafka.common.utils.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link UserTaskManager} keeps track of long-running request.
+ *
+ * {@link HttpServletRequest} can execute for long durations. The servlet submits the asynchronous tasks and returns the
+ * progress of the operation instead of blocking for the operation to complete. {@link UserTaskManager} maintains the
+ * mapping of Request URL and {@link HttpSession} to UserTaskID({@link UUID}). To fetch the status of a request, the
+ * client can use the same Request ULR along with session cookie to retrieve the current status. The status of the
+ * request can also be fetched using the UserTaskID. '/user_tasks' endpoint can be used to fetch all the active and
+ * recently completed UserTasks.
+ */
+public class UserTaskManager implements Closeable {
+  public static final String USER_TASK_HEADER_NAME = "User-Task-ID";
+  public static final long USER_TASK_SCANNER_PERIOD_SECONDS = 5;
+
+  private static final Logger LOG = LoggerFactory.getLogger(UserTaskManager.class);
+  private final Map<SessionKey, UUID> _sessionToUserTaskIdMap;
+  private final Map<UUID, UserTaskInfo> _activeUserTaskIdToFuturesMap;
+  private final Map<UUID, UserTaskInfo> _completedUserTaskIdToFuturesMap;
```

Comment:
Nit: Would using a `SortedMap` for `_activeUserTaskIdToFuturesMap` and `_completedUserTaskIdToFuturesMap` help us improve usability by sorting the tasks based on a comparator ordering `UUID`?